aspnetcore/fundamentals/servers/index.md

```diff
@@ -4,20 +4,32 @@ author: rick-anderson
 description: Discover the web servers Kestrel and HTTP.sys for ASP.NET Core. Learn how to choose a server and when to use a reverse proxy server.
 ms.author: tdykstra
 ms.custom: mvc
-ms.date: 09/13/2018
+ms.date: 09/21/2018
 uid: fundamentals/servers/index
 ---
 # Web server implementations in ASP.NET Core
 
 By [Tom Dykstra](https://github.com/tdykstra), [Steve Smith](https://ardalis.com/), [Stephen Halter](https://twitter.com/halter73), and [Chris Ross](https://github.com/Tratcher)
 
-An ASP.NET Core app runs with an in-process HTTP server implementation. The server implementation listens for HTTP requests and surfaces them to the app as sets of [request features](xref:fundamentals/request-features) composed into an [HttpContext](/dotnet/api/system.web.httpcontext).
+An ASP.NET Core app runs with an in-process HTTP server implementation. The server implementation listens for HTTP requests and surfaces them to the app as sets of [request features](xref:fundamentals/request-features) composed into an <xref:Microsoft.AspNetCore.Http.HttpContext>.
 
-ASP.NET Core ships two server implementations:
+ASP.NET Core ships three server implementations:
```

Comment:
This can be done in a separate PR, but I think we should make inprocess and out of process separate pages.

Reply:
#### TL;DR

I considered it, but it looks like organization, reader flow (navigation), and/or duplication will ensue and make a big mess if the hosting model scenarios are split.

#### Full explanation

I'll describe the landscape and what I ran into.

In ancient times, the team set up the outline like this ...

* Fundamentals
  * Servers - Basic descriptions and links
    * Kestrel - Description and configuration
    * HTTP.sys - Description and configuration
    * ANCM - Basic description only (no config)
* Host-and-deploy ("Publishing" in the olden days) - Basic descriptions and links
  * IIS - Description and configuration
  * ANCM - Configuration

These topics and the content they provide are **_NOT_** easy to arrange.

🐉 **_There be dragons here!_** 🐉

* Kestrel is behind a proxy or on its own. Kestrel is fully baked in; others are more hybrid.
* ANCM/IIS bear some explanation early (*Fundamentals*), but IIS isn't part of .NET Core and fits better in more of a "host/deploy/proxy" set of topics (*Host-and-deploy*).
* IIS has a ton of config.
* ANCM has a lot of separate config.

One thing that could be done is that Kestrel and HTTP.sys could be split between *basic overview* and *configuration*. Kestrel in the *Server* node could merely be an overview (like ANCM/IIS is). Kestrel in the *Host-and-deploy* node could hold the configuration. Same deal for HTTP.sys. That would consistently introduce server concepts in *Fundamentals* but configure every scenario in *Host-and-deploy* (i.e., Kestrel can "host," so it fits; one just doesn't "deploy" anything to it like reverse proxy scenarios). I'd also pitch making that ANCM topic in the *Servers* node *ANCM and IIS* and introduce them with greater balance + see what, if anything, of a general nature in the *Host-and-deploy* IIS topic can be moved there.

Anyway .... *that's a digression!*

I considered splitting ANCM in-proc and out-of-proc, but I wasn't looking at the *Servers* node for that split because the content is so minimal. It's just about three paragraphs of text, a list, and an image. If in-proc and out-of-proc are split out, the reader flow is ...

1. Servers landing
2. In-proc page
3. Back to Servers landing
4. Out-of-proc page
5. Back to Servers landing

... mmm ... 🤔 ... not so good.

The other spot to consider for a split is with the configuration in the *Host-and-deploy* node.

There's no natural way to split the hosting models without creating one of two problems ...

* A lot of duplicated content.
* An ugly navigation flow for the reader (same as above :point_up:)

If configuration is split out for the two hosting models with common config in an ANCM landing page, then the outline is ...

* host-and-deploy/ancm/index - Common config
  * host-and-deploy/ancm/in-proc - In-proc config
  * host-and-deploy/ancm/out-of-proc - Out-of-proc config

**_BUT_** consider ...

* That landing page doesn't explain in-proc/out-of-proc config (it just links). The reader has to navigate in and out of the other two topics, then they nav back to the common config in the landing page for the rest of the ANCM config.
* The landing page either skips the `hostingModel` attribute and pulls it from the attributes table (bad) or the attribute table gets duplicated across the in-proc and out-of-proc topics (also bad; especially bad because in-proc and out-of-proc topics start to main common content).

I didn't see a good workable outline for either the *Servers* node or the *Host-and-deploy* node. Therefore as you see, I in-lined the content in both spots.

Do you have a different outline/approach in mind?

Reply:
I agree that duplicate content/ navigation flow makes the change confusing. This organization seems fine then 😄